pulp_ansible/app/serializers.py

```diff
@@ -515,13 +515,31 @@ class CollectionVersionSignatureSerializer(NoArtifactContentSerializer):
     signing_service = RelatedField(
         help_text=_("The signing service used to create the signature."),
         view_name="signing-services-detail",
-        queryset=SigningService.objects.all(),
+        read_only=True,
         allow_null=True,
     )
 
+    def validate(self, data):
+        """
+        Prepare the necessary data for signature creation. Minimal validation is performed.
+        """
+        data = super().validate(data)
+
+        if "file" not in data:
+            raise ValidationError(_("'file' must be specified."))
```

Comment:
Can't we just mark the file filed as required, and let drf do the tedious part?

Reply:
line 532 explains why this can't rely on DRF, the validate is called outside the request context a second time.

Maybe this `if` must go inside that `if ¨request"` condition.